apps/serdes/Deserializer.cpp

```diff
@@ -0,0 +1,608 @@
+#include "Deserializer.h"
+#include <fstream>
+#include <iostream>
+
+std::string Deserializer::deserialize_string(const flatbuffers::String *str) {
+    return str->str();
+}
+
+Halide::Type Deserializer::deserialize_type(const Halide::Serdes::Type *type) {
+    // bits
+    int bits = type->bits();
+
+    // lanes
+    int lanes = type->lanes();
+
+    // code
+    Halide::Serdes::TypeCode code_deserialized = type->code();
+
+    halide_type_code_t code = halide_type_uint;
+    switch (code_deserialized) {
+    case Halide::Serdes::TypeCode::TypeCode_Int:
```

Comment:
You could make these switches easier to read by putting `using Halide::Serdes::TypeCode;` at the top of this file, then just referencing `TypeCode` directly here (and elsewhere).